create-customer/src/test/java/no/unit/nva/customer/create/CreateCustomerHandlerTest.java

```diff
@@ -0,0 +1,96 @@
+package no.unit.nva.customer.create;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.customer.ObjectMapperConfig;
+import no.unit.nva.customer.model.Customer;
+import no.unit.nva.customer.service.CustomerService;
+import nva.commons.handlers.GatewayResponse;
+import nva.commons.utils.Environment;
+import nva.commons.utils.TestLogger;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.Map;
+
+import static nva.commons.handlers.ApiGatewayHandler.ACCESS_CONTROL_ALLOW_ORIGIN;
+import static nva.commons.handlers.ApiGatewayHandler.CONTENT_TYPE;
+import static org.apache.http.HttpHeaders.ACCEPT;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CreateCustomerHandlerTest {
+
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String WILDCARD = "*";
+
+    private ObjectMapper objectMapper = ObjectMapperConfig.objectMapper;
+    private CustomerService customerService;
+    private Environment environment;
+    private CreateCustomerHandler handler;
+    private ByteArrayOutputStream outputStream;
+    private Context context;
+
+    /**
+     * Setting up test environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        customerService = mock(CustomerService.class);
+        environment = mock(Environment.class);
+        when(environment.readEnv("ALLOWED_ORIGIN")).thenReturn("*");
+        handler = new CreateCustomerHandler(customerService, environment);
+        outputStream = new ByteArrayOutputStream();
+        context = mock(Context.class);
+        when(context.getLogger()).thenReturn(new TestLogger());
+    }
+
+    @Test
+    public void createCustomerHandler() throws Exception {
+        Customer customer = new Customer.Builder()
+                .withName("New Customer")
+                .build();
+        when(customerService.createCustomer(customer)).thenReturn(customer);
+
+        Map<String,Object> headers = getRequestHeaders();
+        InputStream inputStream = inputStream(customer, headers);
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<Customer> actual = objectMapper.readValue(
+                outputStream.toByteArray(),
+                GatewayResponse.class);
+
+        assertEquals(HttpStatus.SC_CREATED, actual.getStatusCode());
```

Comment:
Ideally these are 3 separate tests :)

Reply:
Can we add hashCode and equals to GatewayResponse?

Reply:
https://github.com/BIBSYSDEV/nva-commons/pull/28/files